### ISSUE TITLE
Object.assign() function with ProxyObjects as parameters is cause of com.oracle.truffle.api.dsl.UnsupportedSpecializationException: Unexpected values provided for ExportValueNodeGen

### ISSUE THREAD

**Issue by aliaksei-dudko** [oracle/graal#5906]
**Describe GraalVM and your environment:**
 - GraalVM version: **22.3.0**
 - CE or EE: **CE**
 - JDK version: **JDK17**
 - OS and OS Version: **macOS BigSur 11.7.2**
 - Architecture: **x86**
 - The output of `java -Xinternalversion`:
```
OpenJDK 64-Bit Server VM (17.0.5+8-jvmci-22.3-b08) for bsd-amd64 JRE (17.0.5+8-jvmci-22.3-b08), built on Oct 19 2022 01:49:34 by "graal" with clang Apple LLVM 12.0.0 (clang-1200.0.32.29)

```
**Describe the issue**
Js function Object.assign(target, source) with source and target as ProxyObjects where source contains Java String values throws UnsupportedSpecializationException as can not find appropriate specialization for Java String starting from Graal 22.1 version where TruffleString was introduced. 

**Code snippet or code repository that reproduces the issue**
```
// js function
let vehicle_details = hostObject.getVehicleData(); 
let vehicle_color = hostObject.getColorData();

try {

    // assign color using spread syntax
    let vehicle_details_one = {...vehicle_details, ...vehicle_color};
    console.log("vehicle details with color using spread syntax: " + JSON.stringify(vehicle_details_one) + "\n")

    // assign color using for cycle
    let vehicle_details_two = vehicle_details;
    for (let field in vehicle_color) {
        vehicle_details_two[field] = vehicle_color[field]
    }
    console.log("vehicle details with color using for cycle: " + JSON.stringify(vehicle_details_two) + "\n")

    // assign to new field
    vehicle_details.new_field = vehicle_color;
    console.log("vehicle details with color using new field: " + JSON.stringify(vehicle_details) + "\n")

    **// !!!!ERROR
    // Object.assign
    Object.assign(vehicle_details, vehicle_color); 
    console.log("never reach here");**
} catch (ex) {
    console.log(ex);
}

// host object
public class HostObject {

	@HostAccess.Export
	public ProxyObject getVehicleData() {
		var carData = new HashMap<String, Object>();
		carData.put("costs", 10000);
		carData.put("new", false);
		return ProxyObject.fromMap(carData);
	}

	@HostAccess.Export
	public ProxyObject getColorData() {
		var data = new HashMap<String, Object>();
		data.put("color", "red");
		return ProxyObject.fromMap(data);
	}
}


```
full example: [https://github.com/aliaksei-dudko/truffle-string-error](https://github.com/aliaksei-dudko/truffle-string-error)
**Expected behavior**
String values from the source object are successfully assigned to the target.

**Additional context**
Add any other context about the problem here. Specially important are stack traces or log output. Feel free to link to gists or to screenshots if necesary
<details><summary>Details</summary>

```
org.graalvm.polyglot.PolyglotException: com.oracle.truffle.api.dsl.UnsupportedSpecializationException: Unexpected values provided for ExportValueNodeGen@305f031: [red, JSUndefined, false], [String,Nullish,Boolean]
	at com.oracle.truffle.js.nodes.interop.ExportValueNodeGen.executeAndSpecialize(ExportValueNodeGen.java:249)
	at com.oracle.truffle.js.nodes.interop.ExportValueNodeGen.execute(ExportValueNodeGen.java:120)
	at com.oracle.truffle.js.nodes.interop.ExportValueNode.execute(ExportValueNode.java:76)
	at com.oracle.truffle.js.nodes.access.WriteElementNode$ForeignObjectWriteElementTypeCacheNode.executeWithTargetAndIndexUnguarded(WriteElementNode.java:1831)
	at com.oracle.truffle.js.nodes.access.WriteElementNode.executeWithTargetAndIndexAndValue(WriteElementNode.java:441)
	at com.oracle.truffle.js.nodes.access.WriteElementNode.executeWithTargetAndIndexAndValue(WriteElementNode.java:419)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltins$AssignPropertiesNode.doObject(ObjectFunctionBuiltins.java:974)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$AssignPropertiesNodeGen.executeAndSpecialize(ObjectFunctionBuiltinsFactory.java:2893)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$AssignPropertiesNodeGen.executeVoid(ObjectFunctionBuiltinsFactory.java:2821)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltins$ObjectAssignNode.assign(ObjectFunctionBuiltins.java:922)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$ObjectAssignNodeGen.executeAndSpecialize(ObjectFunctionBuiltinsFactory.java:2704)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$ObjectAssignNodeGen.execute(ObjectFunctionBuiltinsFactory.java:2678)
	at com.oracle.truffle.js.nodes.function.FunctionRootNode.executeInRealm(FunctionRootNode.java:149)
	at com.oracle.truffle.js.runtime.JavaScriptRealmBoundaryRootNode.execute(JavaScriptRealmBoundaryRootNode.java:88)
	at <js> :program(function:5:113-157)
	at org.graalvm.sdk/org.graalvm.polyglot.Context.eval(Context.java:399)
	at org.example.Main.main(Main.java:23)
Original Internal Error: 
com.oracle.truffle.api.dsl.UnsupportedSpecializationException: Unexpected values provided for ExportValueNodeGen@305f031: [red, JSUndefined, false], [String,Nullish,Boolean]
	at com.oracle.truffle.js.nodes.interop.ExportValueNodeGen.executeAndSpecialize(ExportValueNodeGen.java:249)
	at com.oracle.truffle.js.nodes.interop.ExportValueNodeGen.execute(ExportValueNodeGen.java:120)
	at com.oracle.truffle.js.nodes.interop.ExportValueNode.execute(ExportValueNode.java:76)
	at com.oracle.truffle.js.nodes.access.WriteElementNode$ForeignObjectWriteElementTypeCacheNode.executeWithTargetAndIndexUnguarded(WriteElementNode.java:1831)
	at com.oracle.truffle.js.nodes.access.WriteElementNode.executeWithTargetAndIndexAndValue(WriteElementNode.java:441)
	at com.oracle.truffle.js.nodes.access.WriteElementNode.executeWithTargetAndIndexAndValue(WriteElementNode.java:419)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltins$AssignPropertiesNode.doObject(ObjectFunctionBuiltins.java:974)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$AssignPropertiesNodeGen.executeAndSpecialize(ObjectFunctionBuiltinsFactory.java:2893)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$AssignPropertiesNodeGen.executeVoid(ObjectFunctionBuiltinsFactory.java:2821)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltins$ObjectAssignNode.assign(ObjectFunctionBuiltins.java:922)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$ObjectAssignNodeGen.executeAndSpecialize(ObjectFunctionBuiltinsFactory.java:2704)
	at com.oracle.truffle.js.builtins.ObjectFunctionBuiltinsFactory$ObjectAssignNodeGen.execute(ObjectFunctionBuiltinsFactory.java:2678)
	at com.oracle.truffle.js.nodes.function.FunctionRootNode.executeInRealm(FunctionRootNode.java:149)
	at com.oracle.truffle.js.runtime.JavaScriptRealmBoundaryRootNode.execute(JavaScriptRealmBoundaryRootNode.java:88)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:495)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:68)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode$UnboundJSFunctionCacheNode.executeCall(JSFunctionCallNode.java:1314)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode.executeAndSpecialize(JSFunctionCallNode.java:306)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode.executeCall(JSFunctionCallNode.java:251)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode$InvokeNode.execute(JSFunctionCallNode.java:740)
	at com.oracle.truffle.js.nodes.JavaScriptNode.executeVoid(JavaScriptNode.java:192)
	at com.oracle.truffle.js.nodes.binary.DualNode.executeVoid(DualNode.java:142)
	at com.oracle.truffle.js.nodes.control.TryCatchNode.executeVoid(TryCatchNode.java:154)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:80)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:55)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedBlockNode.executeGeneric(OptimizedBlockNode.java:78)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.execute(AbstractBlockNode.java:75)
	at com.oracle.truffle.js.nodes.access.JSWriteCurrentFrameSlotNodeGen.execute_generic4(JSWriteCurrentFrameSlotNodeGen.java:124)
	at com.oracle.truffle.js.nodes.access.JSWriteCurrentFrameSlotNodeGen.execute(JSWriteCurrentFrameSlotNodeGen.java:43)
	at com.oracle.truffle.js.nodes.access.JSWriteCurrentFrameSlotNodeGen.executeVoid(JSWriteCurrentFrameSlotNodeGen.java:330)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:80)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:55)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedBlockNode.executeGeneric(OptimizedBlockNode.java:78)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.execute(AbstractBlockNode.java:75)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeGeneric(AbstractBlockNode.java:85)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeGeneric(AbstractBlockNode.java:55)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedBlockNode.executeGeneric(OptimizedBlockNode.java:80)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.execute(AbstractBlockNode.java:75)
	at com.oracle.truffle.js.nodes.function.FunctionBodyNode.execute(FunctionBodyNode.java:73)
	at com.oracle.truffle.js.nodes.function.FunctionRootNode.executeInRealm(FunctionRootNode.java:149)
	at com.oracle.truffle.js.runtime.JavaScriptRealmBoundaryRootNode.execute(JavaScriptRealmBoundaryRootNode.java:88)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:495)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:68)
	at com.oracle.truffle.js.lang.JavaScriptLanguage$ParsedProgramRoot.execute(JavaScriptLanguage.java:245)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callIndirect(OptimizedCallTarget.java:477)
	at jdk.internal.vm.compiler/org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.call(OptimizedCallTarget.java:458)
	at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotContextImpl.eval(PolyglotContextImpl.java:1470)
	at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotContextDispatch.eval(PolyglotContextDispatch.java:63)
	at org.graalvm.sdk/org.graalvm.polyglot.Context.eval(Context.java:399)

```
</details>

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reaching out, we'll take a look into this shortly

**Comment by oubidar-Abderrahim** [comment]
Tracked internally on GR 44209

**Comment by oubidar-Abderrahim** [comment]
This issue was fixed in 7541f6d9e5a17549acfaa24fe1f3514777e248e8 on GraalJS
